### PR TITLE
[#35] Add scheduled delphinet run

### LINKED ISSUE #35: Run tests in delphinet on schedule
# Clarification and motivation

As result of #34 we now run tests in local chain, but we also would like to run them in delphinet. Carthagenet is probably not very important now because Delphi should come to mainnet in 10 days AFAIU.

# Acceptance criteria

CI runs `baseDAO-test` in delphinet network on schedule. By default we do it once a day, but it's not required (e. g. we can do once per 2 days). It should run on our agent that we use specifically for scheduled nettests.

### PR DESCRIPTION
**Pull request by rvem**
## Description
Problem: We're now running tests on local-chain. However, we should
also do it on real test network.

Solution: Add scheduled run on delphinet.

TODO:
~* change target branch in https://buildkite.com/serokell/basedao/settings/schedules/826742f8-378d-4776-bc3d-cb5f5a09b8d1~
~* add a sane timeout for scheduled job~

## Related issue(s)

Resolves #35 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

**Comment by gromakovsky** [comment]
Rebase should help.

**Comment by gromakovsky** [comment]
Okay, rebased myself, now it should be fine and deserves my approval, but I'll wait for the timeout addition. Let's see if the job will pass.

**Comment by rvem** [comment]
And the tests failed :pled:

**Comment by gromakovsky** [comment]
Apparently it's one more time-sensitive test, we don't handle them well on real network yet. It will be fixed in #47. For now can you please make that test run only on emulator like we did in #44? And let's set timeout to something like 8 hours, so that tests don't get broken immediately after we add just a couple of new tests.

**Comment by gromakovsky** [comment]
Also a conflict was brought that you are welcome to enjoy.

**Comment by rvem** [comment]
Updated, you're welcome to rereview

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -49,19 +57,38 @@ steps:
...
+    commands:
+    - nix-build ci.nix -A packages.baseDAO.tests.baseDAO-test
+    # Emulator + local chain for Delphi protocol
+    - nix run -f ci.nix tezos-client -c
+      ./result/bin/baseDAO-test --nettest-run-network
+      -A delphi.testnet.tezos.serokell.team -P 8732
+    retry:
+      automatic:
+        limit: 1
+    # :so-big:
```
Comment (gromakovsky) [inline .buildkite/pipeline.yml:86]: :D

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -49,19 +57,38 @@ steps:
...
     - nix-build ci.nix -A haddock --no-out-link
 
+  - label: scheduled delphinet test
+    if: build.source == "schedule"
+    # use another agent for long scheduled jobs
+    agents:
+      queue: "scheduled"
+    commands:
+    - nix-build ci.nix -A packages.baseDAO.tests.baseDAO-test
+    # Emulator + local chain for Delphi protocol
```
Comment (gromakovsky) [inline .buildkite/pipeline.yml:79]: Aren't you lying here?

**Review by gromakovsky — APPROVED** [review]
One comment looks like some dirty lies. Hopefully you'll manage to fix it :pled:

And let's wait for CI to pass.

**Comment by gromakovsky** [comment]
CI passed :tada: 

https://buildkite.com/serokell/basedao/builds/380#c0ff11f5-b226-4cbd-936b-6d7e7f537efd

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (rvem) [reply .buildkite/pipeline.yml:79]: There are emulator based tests

Comment (rvem) [reply .buildkite/pipeline.yml:79]: ah, local-chain :pled:

**Pull request merged**